torch_geometric/data/materialized_graph.py

```diff
@@ -0,0 +1,125 @@
+from abc import abstractmethod
+from collections.abc import MutableMapping
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from torch import LongTensor
+
+from torch_geometric.typing import EdgeTensorType, EdgeType
+from torch_geometric.utils.mixin import CastMixin
+
+
+class EdgeLayout(Enum):
+    COO = 'COO'
+    CSC = 'CSC'
+    CSR = 'CSR'
+    LIL = 'LIL'
+
+
+@dataclass
+class EdgeAttr(CastMixin):
+    r"""Defines the attributes of an :obj:`MaterializedGraph` edge."""
+
+    # The layout of the edge representation
+    layout: Optional[EdgeLayout] = None
```

Comment:
does it make sense for this to be optional? if not provided is it a valid `EdgeAttr`?

Reply:
+1. With this as non-optional you can avoid checking `edge_attr` in `put_edge_index`.

Reply:
The reason behind marking this as optional earlier was that, in an earlier design, `layout` was not required for `get_edge_index`. But this has since changed, and you are correct that it no longer makes sense to have `layout` be optional. Updating to reflect this.